app-frontend/src/app/components/tools/diagramContainer/diagramContainer.controller.js

```diff
@@ -51,29 +51,141 @@ export default class DiagramContainerController {
                   <rf-diagram-node-header
                     data-model="model"
                     data-invalid="model.get('invalid')"
-                    data-menu-options="menuOptions"
                   ></rf-diagram-node-header>
+                  <div class="node-actions">
+                    <div class="node-button-group">
+                      <button class="btn node-button" type="button"
+                              ng-if="onPreview"
+                              ng-click="onPreview($event, this.model)">
+                      <span class="icon-eye"></span>
+                      </button>
+                      <button class="btn node-button" type="button"
+                              ng-class="{'active': showHistogram}"
+                              ng-click="toggleHistogram()">
```

Comment:
This shouldn't be available for all node types; if I click this for constant node types, I get a cascade of console errors. Same for the preview button, above.